modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessagingEngine.java

```diff
@@ -280,16 +304,8 @@ public void deleteMessagesById(List<Long> messagesToRemove) throws AndesExceptio
 
 
     public void moveMessageToDeadLetterChannel(List<DeliverableAndesMetadata> messagesToMove) throws AndesException {
-        Map<String, List<AndesMessageMetadata>> storageSeparatedMessages = new HashMap<>();
-        for (DeliverableAndesMetadata message : messagesToMove) {
-            List<AndesMessageMetadata> messagesOfStorageQueue = storageSeparatedMessages
-                    .get(message.getStorageQueueName());
-            if (null == messagesOfStorageQueue) {
-                messagesOfStorageQueue = new ArrayList<>();
-            }
-            messagesOfStorageQueue.add(message);
-            storageSeparatedMessages.put(message.getStorageQueueName(), messagesOfStorageQueue);
-        }
+        Map<String, List<AndesMessageMetadata>> storageSeparatedMessages = groupByStorageQueue(
```

Comment:
I think it's not needed to group by storage queues. IIRC, we move the messages to DLC using the message id anyways.